src/test/scala/com/microsoft/hyperspace/index/rules/RuleTestHelper.scala

```diff
@@ -16,10 +16,17 @@
 
 package com.microsoft.hyperspace.index.rules
 
+import org.apache.hadoop.fs.Path
+import org.apache.spark.sql.SparkSession
+import org.apache.spark.sql.catalyst.expressions.{Attribute, AttributeReference}
 import org.apache.spark.sql.catalyst.plans.logical.LogicalPlan
-import org.apache.spark.sql.execution.datasources.{HadoopFsRelation, LogicalRelation}
+import org.apache.spark.sql.execution.datasources.{FileIndex, HadoopFsRelation, LogicalRelation}
+import org.apache.spark.sql.execution.datasources.parquet.ParquetFileFormat
+import org.apache.spark.sql.types.{StructField, StructType}
 
-import com.microsoft.hyperspace.index.LogicalPlanSignatureProvider
+import com.microsoft.hyperspace.actions.Constants
+import com.microsoft.hyperspace.index.{Content, CoveringIndex, Hdfs, IndexConstants, IndexLogEntry, IndexLogManagerImpl, LogicalPlanFingerprint, LogicalPlanSignatureProvider, NoOpFingerprint, Signature, Source, SparkPlan}
+import com.microsoft.hyperspace.index.serde.LogicalPlanSerDeUtils
 
 object RuleTestHelper {
```

Comment:
Thanks for refactoring these as well.

Can we change this to `trait RuleTestUtils` simliar to [SQLTestUtils](https://github.com/apache/spark/blob/master/sql/core/src/test/scala/org/apache/spark/sql/test/SQLTestUtils.scala) so that you can do `class JoinIndexRuleTest extends HyperspaceSuite with RuleTestUtils`? (and now you can omit `RuleTestHelper.` for most cases).